tests/app/main/views/test_users.py

```diff
@@ -337,6 +340,39 @@ def test_download_csv(self, data_api_client):
         assert response.status_code == 200
         self._assert_things_about_user_export(response, users)
 
+    def test_download_csv_for_on_framework_only(self, data_api_client):
+        framework = self._valid_framework
+        users = [
+            {
+                "application_result": "fail",
+                "application_status": "no_application",
+                "declaration_status": "unstarted",
+                "framework_agreement": False,
+                "supplier_id": 1,
+                "email address": "test.user@sme.com",
+                "user_name": "Test User",
+                "variations_agreed": "var1"
+            },
+            {
+                "application_result": "pass",
+                "application_status": "application",
+                "declaration_status": "complete",
+                "framework_agreement": False,
+                "supplier_id": 2,
+                "email address": "test.user@sme2.com",
+                "user_name": "Test User 2",
+                "variations_agreed": ""
+            }
+        ]
+
+        response = self._return_user_export_response(data_api_client, framework, users, only_on_fwk=True)
+        assert response.status_code == 200
+        rows = [line.split(",") for line in response.get_data(as_text=True).splitlines()]
+
+        assert len(rows) == 2
+        assert rows[0] == ["email address", "user_name", "supplier_id"]
+        assert rows[1] == ["test.user@sme2.com", "Test User 2", "2"]
```

Comment:
This test looks fine, but the two above it seem like they could be combined? Or if we want to keep the filetype/name test separate from the content test, could we have the assertions on `test_download_csv` match those on the new test, so it's clear what the differences are between it and the `only_on_fwk` case?